src/iothread.c

```diff
@@ -444,28 +457,50 @@ void processClientsOfAllIOThreads(void) {
         processClientsFromIOThread(&IOThreads[i]);
     }
 }
+/* Checks all clients in the IO thread for timeouts and buffer resizing needs.
+ * 
+ * For each client:
+ * 1. First checks if client has exceeded max idle time (if configured)
+ *    - Handles timeout via checkClientsHandleTimeout() if needed
+ * 2. Otherwise checks if buffer needs resizing (every 200ms minimum interval)
+ *    - Performs resize via clientsResizeBuffer()
+ *    - Marks with CLIENT_BUFFER_CHANGE flag if buffer was modified
+ *    - Enqueues to main thread for further processing if changes detected
+ * 3. Breaks loop early if neither condition applies (optimization) */
+void checkClients(IOThread *t, mstime_t now) {
+    listIter li;
+    listRewind(t->clients, &li);
+    listNode *ln = listNext(&li);
+    while(ln != NULL) {
+        client *c = listNodeValue(ln);
+        ln = listNext(&li);
+        /* Check for idle timeout first */
+        if (server.maxidletime && (now / 1000) - c->lastinteraction > server.maxidletime) {
+            if (checkClientsHandleTimeout(c, now)) continue;
+        }
+        /* Check buffer resize (min 200ms interval) */
+        else if (now - c->resize_buffer_time > 200) {
+            c->resize_buffer_time = now;
+            /* Try resize and check if changed */
+            if (!clientsResizeBuffer(c, now) && !(c->flags & CLIENT_BUFFER_CHANGE)) continue;
```

Comment:
seems weird, client cron is not just about resizing buffers, does the current change mean that if the client's buffers are not changed, other cron will not take effect?